packages/buckets/src/api/index.ts

```diff
@@ -512,21 +536,33 @@ export async function bucketsPushPath(
   key: string,
   path: string,
   input: any,
-  opts?: { progress?: (num?: number) => void },
+  opts?: PushOptions,
   ctx?: ContextInterface,
 ) {
   return new Promise<PushPathResult>(async (resolve, reject) => {
-    // Only process the first  input if there are more than one
+    // Only process the first input if there are more than one
     const source: File | undefined = (await normaliseInput(input).next()).value
     const client = grpc.client<PushPathRequest, PushPathResponse, APIServicePushPath>(APIService.PushPath, {
       host: api.serviceHost,
       transport: api.rpcOptions.transport,
       debug: api.rpcOptions.debug,
     })
+    // Send a close event to the bucket api upon abort
```

Comment:
So, we listen to the abort event on the signal, and simply close the connection and reject the promise with an AbortError.